File: AndroidSelfLearningTutorials/app/src/main/java/com/example/androidselflearningtutorials/MainActivity.java
package com.example.androidselflearningtutorials;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity// implements View.OnClickListener
{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    public void CallMenuActivity(View view)
    {
        Intent menuIntent = new Intent(this, InitialLoadPage.class);
        startActivity(menuIntent);
    }

//    @Override
//    public void onClick(View view) {
//        switch (view.getId()) {
//            case R.id.enter_menu_button:
//                CallMenuActivity();
//                break;
//            default:
//                break;
//        }
//    }
}

File: NYTimes/app/src/main/java/com/example/nytimes/MainActivity.kt
package com.example.nytimes

import android.Manifest
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import com.nytimes.uiactivities.IntroPageActivity

class MainActivity : AppCompatActivity()
{
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val PERMISSION_ALL = 1
        val PERMISSIONS = arrayOf<String>(
            Manifest.permission.INTERNET,
            Manifest.permission.ACCESS_NETWORK_STATE
        )

        if (!hasPermissions(this, PERMISSIONS)) {
            ActivityCompat.requestPermissions(this, PERMISSIONS, PERMISSION_ALL)
        }
    }


    fun hasPermissions(context: Context?, permissions: Array<String>): Boolean {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && context != null && permissions != null) {
            for (permission in permissions) {
                if (ActivityCompat.checkSelfPermission(
                        context,
                        permission
                    ) != PackageManager.PERMISSION_GRANTED
                ) {
                    return false
                }
            }
        }
        return true
    }

    fun onButtonClicked(view: View) = try
    {
        val introPageIntent: Intent = Intent(this, IntroPageActivity::class.java)
        startActivity(introPageIntent);
    }
    catch (ex: Exception)
    {

    }
}
